### PR TITLE
[CI] Create docker_build.yml

### LINKED ISSUE #4: Auto-build docker image on push
Until we get rid of dataget again for another solution, we need to keep the docker image up to date. To avoid confusion on who has last pushed this and from what, we should set up the same automation here that we already use for other repos.

### PR DESCRIPTION
**Pull request by surchs**
Closes #4 

Changes proposed in this pull request:

- add GH CI workflow to build docker image on push to main
- taken directly from https://github.com/neurobagel/api/blob/main/.github/workflows/ci.yaml

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .github/workflows/docker_build.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  update_docker:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
```
Comment (alyssadai) [inline .github/workflows/docker_build.yml]:
```suggestion
      - name: Checkout
```

File: .github/workflows/docker_build.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        uses: actions/checkout@v3
+      -
+        name: Login to Docker Hub
```
Comment (alyssadai) [inline .github/workflows/docker_build.yml]:
```suggestion
      - name: Login to Docker Hub
```

File: .github/workflows/docker_build.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        name: Set up Docker Buildx
```
Comment (alyssadai) [inline .github/workflows/docker_build.yml]:
```suggestion
      - name: Set up Docker Buildx
```

File: .github/workflows/docker_build.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        uses: docker/setup-buildx-action@v2
+      -
+        name: Build and push
```
Comment (alyssadai) [inline .github/workflows/docker_build.yml]:
```suggestion
      - name: Build and push
```

**Review by alyssadai — APPROVED** [review]
Looks good! I added a few suggestions to fix wonky spacing leftover from the original workflow, otherwise 🧑‍🍳

EDIT: Don't forget to add a PR prefix :)

**Comment by surchs** [comment]
Aha, good points! We should change the other workflow as well

**Pull request merged**